liota/dccs/iotcv2.py

```diff
@@ -0,0 +1,346 @@
+        thread.start()
+
+    def register(self, entity_obj):
+        if isinstance(entity_obj, Metric):
+            # reg_entity_id should be parent's one: not known here yet
+            # will add in creat_relationship();
+            log.info("Registering metric {0} locally".format(entity_obj.name))
+            return RegisteredMetric(entity_obj, self, None)
+        else:
+            log.info("Registering resource with IoTCV2 {0}".format(entity_obj.name))
+            # register fun internal queue
+            reg_resp_q = Queue.Queue()
+
+            def on_response(msg, reg_resp_q):
+                try:
+                    log.debug("Received msg: {0}".format(msg))
+                    json_msg = json.loads(msg)
+                    log.debug("Processing msg: {0}".format(json_msg["type"]))
+                    if json_msg["type"] == "registerMO_rsp" and json_msg["guid"] != "null" and \
+                                    json_msg["guid"] == entity_obj.entity_id:
+                        log.info("Received registration response for: {0}".format(json_msg["guid"]))
+                        if json_msg["result"] == 'SUCCESS':
+                            self.reg_entity_id = json_msg["guid"]
+                    else:
+                        log.info("Waiting for resource registration response")
+                        on_response(reg_resp_q.get(True, timeout), reg_resp_q)
+                except:
+                    raise Exception("Exception while registering resource")
+
+            # create and add register request into req_list (waiting list)
+            operation_id = self.next_id()
+            req = Request(entity_obj.entity_id, operation_id, reg_resp_q)
+            log.info("operation_id:{0} req:{1}".format(operation_id, req))
+            with self.req_ops_lock:
+                self.req_dict.update({str(operation_id): req})
+                log.info("operation_id:{0} req_dict:{1}".format(operation_id, self.req_dict))
+            self.comms.send(json.dumps(
+                self._registration(operation_id, entity_obj.entity_id, entity_obj.name, entity_obj.entity_type)))
+            # block until there is an item available or timeout
+            try:
+                on_response(reg_resp_q.get(True, timeout), reg_resp_q)
+            except:
+                raise Exception("Exception while registering resource")
+
+            if not self.reg_entity_id:
+                raise RegistrationFailure()
+            log.info("Resource Registered {0}".format(entity_obj.name))
+
+            return RegisteredEntity(entity_obj, self, self.reg_entity_id)
+
+    def create_relationship(self, reg_entity_parent, reg_entity_child):
+        # sanity check: must be RegisteredEntity or RegisteredMetricRegisteredMetric
+        if (not isinstance(reg_entity_parent, RegisteredEntity) \
+                    and not isinstance(reg_entity_parent, RegisteredMetric)) \
```

Comment:
Can reg_entity_parent be RegisteredMetric?

Reply:
so far, in v2, No.